Content.Server/Atmos/Piping/Unary/Components/GasThermoMachineComponent.cs

```diff
@@ -30,8 +29,31 @@ public sealed class GasThermoMachineComponent : Component
         [ViewVariables(VVAccess.ReadWrite)]
         public float TargetTemperature = Atmospherics.T20C;
 
-        [DataField("mode")]
-        public ThermoMachineMode Mode = ThermoMachineMode.Freezer;
+        /// <summary>
+        ///     Tolerance for temperature setpoint hysteresis.
+        /// </summary>
+        public float TemperatureTolerance = 2f;
+
+        /// <summary>
+        ///     Implements setpoint hysteresis to prevent heater from rapidly cycling on and off at setpoint.
+        ///     If true, add Sign(Cp)*TemperatureTolerance to the temperature setpoint.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadOnly)]
+        public bool HysteresisState = false;
+
+        public ThermoMachineMode Mode {
+            get
+            {
+                return Cp < 0 ? ThermoMachineMode.Freezer : ThermoMachineMode.Heater;
+            }
+        }
+
+        /// <summary>
+        ///     Coefficient of performance. Output power / input power.
+        ///     Positive for heaters, negative for freezers.
+        /// </summary>
+        [DataField("cp")]
+        public float Cp = 0.9f; // output power / input power, positive is heat
```

Comment:
This is a standard physics thing: https://en.wikipedia.org/wiki/Coefficient_of_performance

I can expand the acronym in the comment in case folks are confused. While excessive jargon should be avoided, there's a reason not every use of IoCManager is expanded to InversionOfControlManager.

Per Mirror's past review feedback I can change the field name in YAML to something more obvious.